yb-voyager/cmd/archiveChangesCommand.go

```diff
@@ -0,0 +1,243 @@
+			continue
+		}
+		segmentsToBeDeleted, err := metaDB.GetSegmentsToBeDeleted()
+		if err != nil {
+			return fmt.Errorf("get segments to be deleted: %v", err)
+		}
+		for _, segment := range segmentsToBeDeleted {
+			err := d.deleteSegment(segment)
+			if err != nil {
+				return fmt.Errorf("delete segment %s: %v", segment.FilePath, err)
+			}
+			if !d.isFSUtilisationExceeded() {
+				break
+			}
+		}
+	}
+	return nil
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+type EventSegmentCopier struct {
+	Dest string
+}
+
+type Segment struct {
+	Num      int
+	FilePath string
+}
+
+func NewEventSegmentCopier(dest string) *EventSegmentCopier {
+	return &EventSegmentCopier{
+		Dest: dest,
+	}
+}
+
+func (m *EventSegmentCopier) getImportCount() (int, error) {
+	msr, err := GetMigrationStatusRecord()
+	if err != nil {
+		return 0, fmt.Errorf("get migration status record: %v", err)
+	}
+	if msr == nil {
+		return 0, fmt.Errorf("migration status record not found")
+	}
+	if msr.FallForwarDBExists {
+		return 2, nil
+	}
+	return 1, nil
+}
+
+func (m *EventSegmentCopier) ifExistsDeleteSegmentFileFromArchive(segmentNewPath string) error {
+	if utils.FileOrFolderExists(segmentNewPath) {
+		err := os.Remove(segmentNewPath)
+		if err != nil {
+			return fmt.Errorf("delete %s: %w", segmentNewPath, err)
+		}
+		log.Infof("Deleted existing file %s", segmentNewPath)
+	}
+	return nil
+}
+
+func (m *EventSegmentCopier) copySegmentFile(segment Segment, segmentNewPath string) error {
+	sourceFile, err := os.Open(segment.FilePath)
+	if err != nil {
+		return fmt.Errorf("open segment file %s : %v", segment.FilePath, err)
+	}
+	defer sourceFile.Close()
+	destinationFile, err := os.Create(segmentNewPath)
+	if err != nil {
+		return fmt.Errorf("create file %s : %v", segmentNewPath, err)
+	}
+	defer destinationFile.Close()
+	_, err = io.Copy(destinationFile, sourceFile)
+	if err != nil {
+		return fmt.Errorf("copy file %s : %v", segment.FilePath, err)
+	}
+	return nil
+}
+
+func (m *EventSegmentCopier) Run() error {
+	importCount, err := m.getImportCount()
+	if err != nil {
+		return fmt.Errorf("getimport count: %v", err)
+	}
+	log.Infof("Import count: %d", importCount)
+
+	for {
+		segmentsToArchive, err := metaDB.GetSegmentsToBeArchived(importCount)
+		if err != nil {
+			return fmt.Errorf("get segments to be archived: %v", err)
+		}
+		for _, segment := range segmentsToArchive {
+			var segmentNewPath string
+			if m.Dest != "" {
+				segmentFileName := filepath.Base(segment.FilePath)
+				segmentNewPath = fmt.Sprintf("%s/%s", m.Dest, segmentFileName)
+
+				err := m.ifExistsDeleteSegmentFileFromArchive(segmentNewPath)
+				if err != nil {
+					return fmt.Errorf("delete existing file %s : %v", segmentNewPath, err)
+				}
+
+				err = m.copySegmentFile(segment, segmentNewPath)
+				if err != nil {
+					return fmt.Errorf("copy file %s : %v", segment.FilePath, err)
+				}
```

Comment:
Add a PrintAndLog() here telling the segment file <PATH> is copied to the <NEW_PATH>.

Reply:
Fixed